src/freetype/ft-ogl.adb

```diff
@@ -0,0 +1,285 @@
+--------------------------------------------------------------------------------
+-- Copyright (c) 2012, Felix Krause <flyx@isobeef.org>
+--
+-- Permission to use, copy, modify, and/or distribute this software for any
+-- purpose with or without fee is hereby granted, provided that the above
+-- copyright notice and this permission notice appear in all copies.
+--
+-- THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+-- WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+-- MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+-- ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+-- WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+-- ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+-- OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+--------------------------------------------------------------------------------
+
+with Ada.Text_IO; use Ada.Text_IO;
+
+with GL.Attributes;
+with GL.Blending;
+with GL.Objects.Buffers;
+with GL.Objects.Vertex_Arrays;
+with GL.Objects.Textures.Targets;
+with GL.Pixels;
+with GL.Toggles;
+
+with FT.Faces;
+with FT.Glyphs;
+
+package body FT.OGL is
+   use Interfaces.C;
+    procedure Load_Vertex_Buffer is new
+     GL.Objects.Buffers.Load_To_Buffer (GL.Types.Singles.Vector4_Pointers);
+
+   type Character_Data_Vector is array (Natural range <>) of Character_Record;
+
+   Face_Ptr             : FT.Faces.Face_Reference;
+   Vertex_Array         : GL.Objects.Vertex_Arrays.Vertex_Array_Object;
+   Vertex_Buffer        : GL.Objects.Buffers.Buffer;
+   Extended_Ascii_Data  : Character_Data_Vector (0 .. 255);
```

Comment:
I think these global vars should be in a private type `Font_Renderer` instead. Maybe together with the `Face_Reference` above. So the user initialises a `Font_Renderer` object and afterwards can draw text with it.

This will also allow loading multiple fonts at the same time.

Reply:
This sounds an excellent idea.
However, I would prefer to develop it as a separate example if that's OK.
I prefer to preserve the original context of this example to keep it in line with the original C example that it is based on.